webapp/src/js/services/enketo.js

```diff
@@ -349,7 +356,28 @@ angular.module('inboxServices').service('Enketo',
         $(selector).on('valuechange.enketo', listener);
       }
     };
-                        
+
+    var registerValidationCompleteListener = function(selector) {
+      const listener = () => {
+        if ($('.invalid-required').length > 0) {
+          $(selector).find('.next-page').attr('disabled','disabled');
+        } else {
+          $(selector).find('.next-page').removeAttr('disabled');
+        }
```

Comment:
This is crazy similar to the code on line 310 - can you pull it out as a new function and reuse it?